src/geometry/point.rs

```diff
@@ -26,7 +26,7 @@ impl Point {
     }
 
     /// Returns the squared distance from this point to the given one
-    pub fn squared_distance_to(&self, target: &Point) -> f32 {
+    pub fn squared_distance_to(self, target: Point) -> f32 {
```

Comment:
[Clippy](https://github.com/rust-lang-nursery/rust-clippy) stated: "_this argument is passed by reference, but would be more efficient if passed by value_".